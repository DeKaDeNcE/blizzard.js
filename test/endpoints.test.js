const endpoints = require('../lib/endpoints');

describe('lib/endpoints.js', () => {

  test('should export functions', () => {
    expect(endpoints).toEqual(expect.objectContaining({
      getEndpoint: expect.any(Function),
    }));
  });

  describe('#getEndpoint()', () => {

    test('should return the default endpoint', () => {
      const endpoint = endpoints.getEndpoint();
      expect(endpoint).toMatchSnapshot();
    });

    test('should return the US endpoint', () => {
      const endpoint = endpoints.getEndpoint('foo');
      expect(endpoint).toMatchSnapshot();
    });

    test('should return the SEA endpoint', () => {
      const endpoint = endpoints.getEndpoint('sea');
      expect(endpoint).toMatchSnapshot();
    });

    test('should return the EU es_ES endpoint', () => {
      const endpoint = endpoints.getEndpoint('eu', 'es_ES');
      expect(endpoint).toMatchSnapshot();
    });

  });

});
