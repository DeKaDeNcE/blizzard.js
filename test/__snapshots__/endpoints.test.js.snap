// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/endpoints.js #getEndpoint() should return the EU es_ES endpoint 1`] = `
Object {
  "hostname": "https://eu.api.battle.net",
  "locale": "es_ES",
}
`;

exports[`lib/endpoints.js #getEndpoint() should return the SEA endpoint 1`] = `
Object {
  "hostname": "https://sea.api.battle.net",
  "locale": "en_US",
}
`;

exports[`lib/endpoints.js #getEndpoint() should return the US endpoint 1`] = `
Object {
  "hostname": "https://us.api.battle.net",
  "locale": "en_US",
}
`;

exports[`lib/endpoints.js #getEndpoint() should return the default endpoint 1`] = `
Object {
  "hostname": "https://us.api.battle.net",
  "locale": "en_US",
}
`;
